UnityProject/Assets/Scripts/Items/Traits/CommonTraits.cs

```diff
@@ -69,6 +69,7 @@ public class CommonTraits : SingletonScriptableObject<CommonTraits>
 	[BoxGroup("Materials")] public ItemTrait OrePlasma;
 
 	[BoxGroup("Clothing")] public ItemTrait Mask;
+	[BoxGroup("Clothing")] public ItemTrait GasMask;
```

Comment:
Instead of adding this trait, which is rather uncommonly used, to the `CommonTraits` SO, I suggest serializing reference to it directly in `RespiratorySystem`.

Reply:
aighty, will do

Reply:
...actually, there are a bunch of significantly less common items that use itemtraits that are placed in CommonTraits, like "Emag" and "Hatchet," both of which are only used by one item. GasMask is used by a total of 5 items

Reply:
Yes, and it annoys me to no end ; )

Reply:
GasMask is used by 5 items? Are you sure? I could only find two references here.

Reply:
...it occurs to me that I was also the one to add _both_ of those traits... man, i really am a shitcoder huh

Reply:
> GasMask is used by 5 items? Are you sure? I could only find two references here.

i made sure all five gas masks were using the GasMask trait

Reply:
Oh right, but only one script.

Reply:
Speaking of that - do the items not use inheritance? Why can't you get away with assigning it to just the base prefab?

Reply:
they don't use inheritance which is super obnoxious but i didn't feel like changing all of them at the time

Reply:
Ah I see 👍. That's a little annoying for you. All good then.

Reply:
I thought you were going to remove the trait from the common traits list.

Reply:
i'm not entirely sure where to put it

Reply:
Honestly I don't think it is needed at all. You could instead of getting the trait, try to get the gas mask component from the clothing item in gas slot like this ``TryGetComponent<GasMask>(out var gasMask)``. Add all those messages to that component and let the gas mask handle the filtering of the air, so you could change effectiveness of a particular gas mask or custom messages or whatever. I can guide you on discord on doing that.

Reply:
there is no gas mask component

Reply:
i might add that in a later PR but for now i'm just going to be done with this